test/cppsuite/test_harness/workload/thread_context.h

```diff
@@ -157,175 +112,46 @@ class transaction_context {
-          value.c_str(), ts, op_track_cursor);
-        if (ret != 0) {
-            if (ret == WT_ROLLBACK) {
-                transaction.rollback(session.get(), "");
-                return (false);
-            } else
-                testutil_die(
-                  ret, "unhandled error while trying to save an update to the tracking table");
-        }
-        transaction.add_op();
-        return (true);
-    }
+    bool update(scoped_cursor &cursor, uint64_t collection_id, const std::string &key);
 
     /*
      * Generic insert function, takes a collection_id and key_id, will generate the value.
      *
      * Returns true if it successfully inserts the key, false if it receives rollback from the API.
      */
-    bool
-    insert(scoped_cursor &cursor, uint64_t collection_id, uint64_t key_id)
-    {
-        WT_DECL_RET;
-        std::string key, value;
-        testutil_assert(tracking != nullptr);
-        testutil_assert(cursor.get() != nullptr);
+    bool insert(scoped_cursor &cursor, uint64_t collection_id, uint64_t key_id);
 
-        /*
-         * Get a timestamp to apply to the update. We still do this even if timestamps aren't
-         * enabled as it will return a value for the tracking table.
-         */
-        wt_timestamp_t ts = timestamp_manager->get_next_ts();
-        transaction.set_commit_timestamp(session.get(), ts);
-
-        key = key_to_string(key_id);
-        value = random_generator::instance().generate_string(value_size);
-
-        cursor->set_key(cursor.get(), key.c_str());
-        cursor->set_value(cursor.get(), value.c_str());
-        ret = cursor->insert(cursor.get());
-        if (ret != 0) {
-            if (ret == WT_ROLLBACK) {
-                transaction.rollback(session.get(), "");
-                return (false);
-            } else
-                testutil_die(ret, "unhandled error while trying to insert a key");
-        }
-        ret = tracking->save_operation(tracking_operation::INSERT, collection_id, key.c_str(),
-          value.c_str(), ts, op_track_cursor);
-        if (ret != 0) {
-            if (ret == WT_ROLLBACK) {
-                transaction.rollback(session.get(), "");
-                return (false);
-            } else
-                testutil_die(
-                  ret, "unhandled error while trying to save an insert to the tracking table");
-        }
-        transaction.add_op();
-        return (true);
-    }
-
-    void
-    sleep()
-    {
-        _throttle.sleep();
-    }
-
-    bool
-    running() const
-    {
-        return (_running);
-    }
+    void sleep();
+    bool running() const;
 
+    public:
     scoped_session session;
     scoped_cursor op_track_cursor;
     transaction_context transaction;
-    test_harness::timestamp_manager *timestamp_manager;
-    test_harness::workload_tracking *tracking;
-    test_harness::database &database;
+    timestamp_manager *tmstmp_manager;
```

Comment:
I see you renamed timestamp_manager to tsmtmp_manager I don't like this personally, I'd rather have the full name or just `tsm` for `timestamp_manager`. What do others thing?

Reply:
There were compilation issues with `timestamp_manager *timestamp_manager;` and `database &database`. I had to rename them. I like `tsm`.

Reply:
Sure tsm is fine then.